### PR TITLE
Improve WebSockets API

### PR DESCRIPTION
**Pull request by iliajie**
Howdy Jamie!

Since we initially created WebSockets solely for the Terminal module, there were quite a few dependencies involved.

Today, I've been working on addressing all these issues.

Additionally, I've added a new function to get the socket URL based on the configuration. The configuration for the WebSocket base port and host has been moved to the Webmin configuration, where it belongs.

Streamlining the cleanup of no longer used WebSockets is also part of this update.

Please take a look.

### CODE REVIEW

File: web-lib-funcs.pl [hunk 1]
```diff
@@ -13522,6 +13522,139 @@ sub get_lock_links_dir
...
+    }
+&unlock_file(&get_miniserv_config_file());
+}
+
+# get_miniserv_websockets_modules()
+# Returns a list of modules and themes that use websockets
+sub get_miniserv_websockets_modules
+{
+my @rv;
+foreach (&get_all_module_infos(), &list_themes()) {
```
Comment (jcameron) [inline web-lib-funcs.pl:13652]: Use a proper loop variable instead of $_

File: cron/cron-lib.pl [hunk 2]
```diff
@@ -1605,9 +1605,8 @@ sub cleanup_temp_files
 closedir(DIR);
 
 # Cleanup old websockets
-if (&foreign_check("xterm")) {
-	&foreign_require("xterm");
-	&xterm::cleanup_miniserv_websockets();
+foreach (&get_miniserv_websockets_modules) {
```
Comment (jcameron) [inline cron/cron-lib.pl:1608]: Shouldn't this be `&get_miniserv_websockets_modules()`

File: web-lib-funcs.pl (hunk 1 above)
Comment (iliajie) [reply web-lib-funcs.pl:13652]: I had trouble naming this variable nicely, so I decided not to use it at all. Is it really necessary? It's a very simple call.

File: cron/cron-lib.pl (hunk 2 above)
Comment (iliajie) [reply cron/cron-lib.pl]: Oh, you're right! Fixed now! Thanks!

File: web-lib-funcs.pl (hunk 1 above)
Comment (iliajie) [reply web-lib-funcs.pl]: Okay. Fixed now too.

Comment (jcameron) [reply web-lib-funcs.pl]: Thanks! I just dislike the magical $_ perl variable

Comment (iliajie) [reply web-lib-funcs.pl]: Sure! No problem

Comment (jcameron) [reply web-lib-funcs.pl]: Wait, did you actually fix this? I don't see it in the diff..

Comment (iliajie) [reply web-lib-funcs.pl]: > Wait, did you actually fix this? I don't see it in the diff..

https://github.com/webmin/webmin/pull/2191/files#diff-bbebbbbee3314a0e1686dfb6d99107e442573a1d68b87102c51fe94f3cc18b8dR13649-R13656

Comment (jcameron) [reply web-lib-funcs.pl]: Thanks! Sometimes I think github's "Files changed" tab is laggy..

Comment (iliajie) [reply web-lib-funcs.pl]: No problem!

**Pull request merged**